### DISCUSSION TITLE
Install old firmware for Adapter Shield v5

### DISCUSSION THREAD

**Discussion by worstover** [john30/ebusd#1277] [Q&A]
I had the adapter running without any problems for several months from the beginning. But since a few months I have a lot of errors on the bus and only a reboot of the adapter fixes it most of the times. If not a restart is required.

I have erased and reconfigured the adapter but nothing seems to help. As I can't say when the problem arose, I want to install a much older firmware, but in the documentation only the past three actual versions are linked and installable via the web installer.

So my question: Is there a way to download and install older firmware versions for adapter shield v5?

Thanks in advance!

**Comment by LeZerb** [comment]
You could refer to the [changelog](https://adapter.ebusd.eu/v5/ChangeLog) for the version you need and then download the firmware binary and flash it with esptool.

For Version 2023-12-17 the binary would for instance be available at [https://adapter.ebusd.eu/v5/files/20231217.bin](https://adapter.ebusd.eu/v5/files/20231217.bin). There is also some metadata that the web flash tool uses at [https://adapter.ebusd.eu/v5/files/20231217.json](https://adapter.ebusd.eu/v5/files/20231217.json)

The binary needs to be flashed at flash offset 0x10000. There is however also a second slot (for a backup application image) in flash that might be in use if an OTA update was performed after the initial flash. Then the image at 0x10000 might not automatically start.

**Comment by worstover** [comment]
Maybe I did not see it before or it is a new function that has been implemented, but now it's possible even for older versions to flash them via ESP Web Tools. 

Thank you for support!
